src/osx/cocoa/textctrl.mm

```diff
@@ -562,6 +583,29 @@ - (void)cut:(id)sender
 
 - (void)paste:(id)sender
 {
+    NSPasteboard *pb = [NSPasteboard generalPasteboard];
+    NSString *pbItem = [pb readObjectsForClasses: @[[NSString class],[NSAttributedString class]] options:nil].lastObject;
+    if ([pbItem isKindOfClass:[NSAttributedString class]])
+        pbItem = [(NSAttributedString *)pbItem string];
+
+    wxWidgetCocoaImpl* impl = (wxWidgetCocoaImpl* ) wxWidgetImpl::FindFromWXWidget( self );
+    if ( impl )
+    {
+        wxTextCtrl *textCtrl = (wxTextCtrl *) impl->GetWXPeer();
+        if( textCtrl && textCtrl->GetTextPeer() )
+        {
+            if( textCtrl->HasFlag( wxTE_MULTILINE ) )
+            {
+                auto maxlen = textCtrl->GetMaxLen();
+                if( textCtrl->GetValue().length() + pbItem.length > maxlen )
+                {
+                    auto range = maxlen - textCtrl->GetValue().length();
+                    pbItem = [pbItem substringToIndex:range];
+                    [self insertText:pbItem];
```

Comment:
This looks wrong: if we insert text here and call `[super paste:]` below too, the text will be inserted twice (assuming `replacementString:` is not called).